Content.Server/Fluids/EntitySystems/PuddleSystem.cs

```diff
@@ -319,6 +323,26 @@ private void UpdateSlip(EntityUid entityUid, PuddleComponent component, Solution
         }
     }
 
+    private void UpdateSlow(EntityUid uid, Solution solution)
+    {
+        var totalViscosity = FixedPoint2.Zero;
+        foreach (var reagent in solution.Contents)
+        {
+            var reagentProto = _prototypeManager.Index<ReagentPrototype>(reagent.ReagentId);
+            totalViscosity += reagent.Quantity * FixedPoint2.New(reagentProto.Viscosity);
+        }
+        if (totalViscosity > 0)
+        {
+            var comp = EnsureComp<SlowContactsComponent>(uid);
+            var speed = 1 - (totalViscosity / solution.Volume).Float();
+            _slowContacts.ChangeModifiers(uid, speed, comp);
+        }
```

Comment:
This is going to prioritise the last reagent which I don't think should be done, I think it should just use mathf.min.

Reply:
It's an average, e.g. 50% of water and 50% of glue will be 25% slowdown.